tests/chainer_tests/test_variable.py

```diff
@@ -1420,6 +1420,71 @@ def test_bool_inconvertible(self):
                 pass
 
 
+_set_data_backend_params = [
+    # NumPy
+    {},
+    # CuPy
+    {'use_cuda': True, 'cuda_device': 0},
+    {'use_cuda': True, 'cuda_device': 1},
+    # ChainerX
+    {'use_chainerx': True, 'chainerx_device': 'native:0'},
+    {'use_chainerx': True, 'chainerx_device': 'cuda:0'},
+    {'use_chainerx': True, 'chainerx_device': 'cuda:1'},
+]
+
+
+@testing.parameterize(*testing.product(
+    {
+        'x_shape': [(10,), ()],  # TODO(okapies): add tests for None
+        'requires_grads': [
+            (False, False), (True, False), (False, True), (True, True),
+        ],
+    }))
+@testing.backend.inject_backend_tests(None, _set_data_backend_params)
+@testing.backend.inject_backend_tests(None, _set_data_backend_params)
+class TestVariableSetData(unittest.TestCase):
+
+    def _gen_array(self, device, shape, requires_grad):
+        if shape is None:
+            a = None
+        else:
+            a = device.send_array(
+                np.random.uniform(-1, 1, shape).astype(np.float32))
+
+        if a is not None and device.xp is chainerx and requires_grad:
+            return a.require_grad()
+        else:
+            return a
+
+    def test_set_data(self, backend_config1, backend_config2):
+        device_x = backend_config1.device
+        device_y = backend_config2.device
+        xp_x = device_x.xp
+        requires_grad_x = self.requires_grads[0]
+        requires_grad_y = self.requires_grads[1]
+
+        x = self._gen_array(device_x, self.x_shape, requires_grad_x)
+        y = self._gen_array(device_y, self.x_shape, requires_grad_y)
+
+        v = chainer.Variable(x, requires_grad=requires_grad_x)
+
+        should_fail = (
+            (
+                not device_x == device_y
+                and not device_x == intel64.Intel64Device) or
```

Comment:
Do you mean `isinstance`?
Currently this expression should always evaluate to `True`.

Reply:
Good catch, fixed.